### PR TITLE
guard against AttributeError exception in get_device_list

### PR DESCRIPTION
**Pull request by Stricted**
when you provide an empty builds.json
```
[]
```
then the whole updater will crash with the following exception when trying to access it
```
  File "/home/update/updater/app.py", line 82, in get_device_list
    return builds.keys()
AttributeError: 'list' object has no attribute 'keys'
```

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -79,7 +79,10 @@ def get_builds():
         raise UpstreamApiException('Unable to contact upstream API')
 
 def get_device_list():
-    return get_builds().keys()
+    try:
+        return get_builds.keys()
```
Comment (fourkbomb) [inline app.py]: `get_builds()`

**Review by fourkbomb — CHANGES_REQUESTED** [review]
I'm not convinced that this is really a case we need to handle: the actual mirrorbits-api will always return at least an empty dictionary, so probably the "right" fix would just be to make your empty builds.json contain `{}`. 

FWIW: Python's `json` module refuses to load an empty string for me.

**Comment by Stricted** [comment]
im using php to generate the builds.json

thats the way PHP generates an empty json string
https://3v4l.org/Jr3Kd

i just came across this when setting up the updater when no builds where there yet
im fine with abandoning this

**Comment by zifnab06** [comment]
That's because it's not an array - the data structure is a map of device codenames to build blobs. I don't think we need this.

**Pull request closed without merging**